task-driver/integration/tests/update_wallet.rs

```diff
@@ -47,57 +71,233 @@ async fn test_update_wallet_then_recover(test_args: IntegrationTestArgs) -> Resu
 
     let (_task_id, handle) = test_args.driver.start_task(task).await;
     let success = handle.await?;
-    assert_true_result!(success)?;
+    assert_true_result!(success)
+}
 
-    // Now attempt to lookup the wallet and verify its construction
-    lookup_wallet_and_check_result(&wallet, blinder_seed, share_seed, test_args).await
+/// Execute a wallet update, then lookup the new wallet from on-chain state and verify it has
+/// been correctly constructed
+async fn execute_wallet_update_and_verify(
```

Comment:
tiny nit: this name is a bit vague, could we do `execute_wallet_update_and_verify_shares`?